rdr_service/services/gunicorn_config.py

```diff
@@ -23,3 +24,29 @@
     "RDR_CONFIG_PROVIDER={0}".format(os.environ.get('RDR_CONFIG_PROVIDER', None)),
     "RDR_STORAGE_PROVIDER={0}".format(os.environ.get('RDR_STORAGE_PROVIDER', None)),
 ]
+
+
+# GAE F4 instances allow for up to 1G of memory to be used.
+# If we're getting too close to the limit we should close the instance.
+# That way we can gracefully limit our memory rather than have Google killing us forcefully
+# (and potentially in the middle of handling a request).
+def post_request(worker, *_):
+    # Sum up memory used for this process and any children (resulting in memory used by this instance)
+    self_mem_bytes = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
+    children_mem_bytes = resource.getrusage(resource.RUSAGE_CHILDREN).ru_maxrss
+    memory_threshold_bytes = 950000000  # 950 megabytes
+
+    # Gracefully kill the worker that just completed a request if the instance is holding on to too much memory.
+    # This unfortunate worker may not be the one using the most memory on the instance but it will help,
+    # and hopefully we'll eventually get the one that is.
+    # In the future this can be tuned to use the process id of the worker to find how much memory that worker
+    # has allocated.
+    memory_used_bytes = self_mem_bytes + children_mem_bytes
+    if memory_used_bytes > memory_threshold_bytes:
+
+        # This is copied from Gunicorn's code for closing out a sync worker after reaching the max_requests limit
+        worker.alive = False
+
+        # Logs from the worker seem to appear beside the normal app logs, but without log levels attached to them.
+        worker.log.info(f"Auto-restarting worker after gunicorn instance {os.getpid()} was"
```

Comment:
The `post_request` method is called on the worker's thread, and the `alive` field is checked later (after this method to complete) to determine if the worker should continue running. The logger on the worker doesn't seem to check the alive status before processing statements. I think it's safe this way, but I can switch them around if you think it would be better for readability.

Reply:
Just wanted to be sure the `worker.log` operation wasn't affected by whether `worker.alive` attribute was still True.